### PR TITLE
Implement safe deletion strategy

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- add safe deletion strategy and allow to switch between strategies by changing one const 

**Related issue(s)**

https://github.com/kyma-project/keda-manager/issues/89

### CODE REVIEW

**Comment by MichalKalke** [comment]
LGTM

**Review by kwiatekus — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -207,6 +207,8 @@ kubectl apply -f config/samples/operator_v1alpha1_keda_k3d.yaml
 kubectl delete -f config/samples/operator_v1alpha1_keda_k3d.yaml
 ```
 
+> **NOTE:** keda-manager uses a safe deletion strategy to uninstall the keda module from the cluster. It means that the keda-manager would keep all keda's CRDs on the cluster after the uninstallation process.
```
Comment (grego952) [inline README.md:210]:
```suggestion
> **NOTE:** `keda-manager` uses a safe deletion strategy to uninstall the keda module from the cluster. It means that `keda-manager` keeps all Keda's CRDs on the cluster after the uninstallation process.
```

**Review by grego952 — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Comment by kwiatekus** [comment]
Recent agreement in the modularasation sync states as follows:

When a user triggers module deletion, it should only be processed (Module CR removal, undeploying module workloads ) only once user deletes all CRs he created.
The Deletion should be triggered byt blocked until all user created data (CRs) are removed by the user.
We should not leave orphaned CRs. Keda and Kyma CR should reflect the state whenever deletionof a module is blocked by the existance on user created CRs

In Keda case:
User will effectively remove keda module only if we first manuall removes all ScaledObjects created by him.
If he doesnt do it, keda module deletion is initaited but blocked. The remaining user created Scaled objects should remain controlled by the keda workloads. Keda CR status reflects the blocked deletion
Once User removes all the CRs (Scaled objects), the deletion process will continue and there will be nothing left that belongs to keda module on the cluster.

**Pull request merged**